cmd/contour/serve.go

```diff
@@ -396,3 +425,13 @@ func startInformer(inf informer, log logrus.FieldLogger) func(stop <-chan struct
 		return nil
 	}
 }
+
+func startDynamicInformer(inf cache.SharedIndexInformer, log logrus.FieldLogger) func(stop <-chan struct{}) error {
+	return func(stop <-chan struct{}) error {
+		log.Println("starting")
```

Comment:
The informers are different types and have different ways to start. `informer` uses `Start()` and `cache.SharedIndexInformer` uses `Run()`. I'm not sure we can combine into the same method.

Reply:
Ok, this part is extremely confusing.

The `informer` interface actually matches on the `coreinformers.SharedInformerFactory` type, because that type implements `(*internalinterfaces.SharedInformerFactory.Start)`. So we refer to both informers and informer factory as "informers" in the helper functions without being really clear about the differences.

So `dynamicFactory` is an object of type `*dynamic informer.DynamicSharedInformerFactory`, and it is this object that you want to `Start()` using `startInformer(). You only need to start the factory, not each informer that the factory generates.

We can improve this further (probably in a separate PR) by calling `WaitForCacheSync` on the factories rather than on each individual informer.